Sources/Result/ResultType.swift

```diff
@@ -0,0 +1,41 @@
+//
+//  ResultType.swift
+//  Deferred
+//
+//  Created by Zachary Waldowski on 12/9/15.
+//  Copyright © 2014-2016 Big Nerd Ranch. Licensed under MIT.
+//
+
+/// A type that can exclusively represent either some result value of a
+/// successful computation or a failure with an error.
+public protocol ResultType: CustomStringConvertible, CustomDebugStringConvertible {
+    associatedtype Value
```

Comment:
I invite improvements to the name of this associated type. Bike shedders, activate!